### PR TITLE
fix @magentoConfigFixture on website level

### PR DESCRIPTION
**Pull request by yaroslav-zenin**
### Description (*)
In unit test method annotation `@magentoConfigFixture` does not apply to method on website level
```php
    /**
     * @magentoConfigFixture websites/base/modulename/foo/bar value
     */
    public function testSomething()
    {}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php [hunk 1]
```diff
@@ -67,12 +68,12 @@ protected function _getConfigValue($configPath, $scopeCode = null)
      *
      * @param string $configPath
      * @param string $value
-     * @param string|bool|null $storeCode
+     * @param string|bool|null $scopeCode
      */
-    protected function _setConfigValue($configPath, $value, $storeCode = false)
+    protected function _setConfigValue($configPath, $value, $scopeCode = false, $scopeType = ScopeInterface::SCOPE_STORE)
```
Comment (larsroettig) [inline dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php]: Adding parameters in protected methods is not allowed see our 
[**Backward Compatible Guide**](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

File: dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php [hunk 2]
```diff
@@ -46,7 +47,7 @@ class ConfigFixture
      * @param string|bool|null $scopeCode
      * @return string
      */
-    protected function _getConfigValue($configPath, $scopeCode = null)
+    protected function _getConfigValue($configPath, $scopeCode = null, $scopeType = ScopeInterface::SCOPE_STORE)
```
Comment (larsroettig) [inline dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php]: Change parameter type in protected methods is not allowed see our
[**Backward Compatible Guide**](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by larsroettig — CHANGES_REQUESTED** [review]

**Comment by yaroslav-zenin** [comment]
@larsroettig I did change according `Backward Compatible Guide`
I understand that we cannot add extra param to public or protected methods because of declaration compability for children objects.
At the moment for me is not clear if somebody will extend and change behavior of methods `_setConfigValue` or `_getConfigValue` it will not use it, because to use `$scopeType` param I call directly my private functions
how to fix it in right way?

**Comment by sivaschenko** [comment]
@yaroslav-zenin the third party that extends and rewrites `_setConfigValue` or `_getConfigValue` will have an option to consider the changes introduced in this pull request in order to fix the bug. However, with backward compatible approach there will be no pressure to adopt this fix.

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@yaroslav-zenin thanks for the pull request! Can you please add more details to the description to provide a better understanding of the use case and test case to verify the issue and fix.

Also, covering changes with a test would be really handy as the test will provide examples of syntax to use for requesting config fixture on website level).

Finally, I think that a `website` keyword will look more appropriate as part of scope code rather than config path. I.e.: `@magentoConfigFixture website:my_website_code config/path/to/set value`

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22044) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=22044) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: yaroslav-zenin<br/>:x: Yaroslav Zenin<hr/>**Yaroslav Zenin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@yaroslav-zenin Looks like the pull request branch is not properly synced with 2.3-develop. Can you please recreate the pull request branch from magento/magento2 2.3-develop.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@yaroslav-zenin thanks for the update. The CLA checker fails because of the commit author is not properly linked to the GitHub account. Please take a look at the commit authorship, it looks to be different for the last two commits:

image

Could you please ensure the local git is configured using the same email as GitHub account, squash all the commits and force push a sigle commit with a correct author email

**Comment by sivaschenko** [comment]
Hi @yaroslav-zenin can you please ensure your local git is configured using the email associated with you GitHub account, squash all the commits into a single one and force push a single commit with the correct authorship.

Unfortunately, we cannot process this pull request until all the commits will be correctly linked to an account

**Comment by sidolov** [comment]
@yaroslav-zenin , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**